pkg/cmd/roachtest/prometheus/prometheus.go

```diff
@@ -0,0 +1,209 @@
+		ctx,
+		c,
+		cfg.ScrapeConfigs,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := c.PutString(
+		ctx,
+		yamlCfg,
+		"/tmp/prometheus/prometheus.yml",
+		0644,
+		cfg.PrometheusNode,
+	); err != nil {
+		return nil, err
+	}
+
+	// Start prometheus as systemd.
+	if err := c.RunE(
+		ctx,
+		cfg.PrometheusNode,
+		`cd /tmp/prometheus &&
+sudo systemd-run --unit prometheus --same-dir \
+	./prometheus --config.file=prometheus.yml --storage.tsdb.path=data/ --web.enable-admin-api`,
+	); err != nil {
+		return nil, err
+	}
+	return &Prometheus{Config: cfg}, nil
+}
+
+// Snapshot takes a snapshot of prometheus and stores the snapshot in the given localPath
+func (pm *Prometheus) Snapshot(
+	ctx context.Context, c cluster, l *logger.Logger, localPath string,
+) error {
+	if err := c.RunE(
+		ctx,
+		pm.PrometheusNode,
+		`curl -XPOST http://localhost:9090/api/v1/admin/tsdb/snapshot &&
+	cd /tmp/prometheus && tar cvf prometheus-snapshot.tar.gz data/snapshots`,
+	); err != nil {
+		return err
+	}
+
+	return c.Get(
+		ctx,
+		l,
+		"/tmp/prometheus/prometheus-snapshot.tar.gz",
+		localPath,
+		pm.PrometheusNode,
+	)
+}
+
+// makeYAMLConfig creates a prometheus YAML config for the server to use.
+func makeYAMLConfig(ctx context.Context, c cluster, scrapeConfigs []ScrapeConfig) (string, error) {
+	type yamlStaticConfig struct {
+		Targets []string
+	}
+
+	type yamlScrapeConfig struct {
+		JobName       string             `yaml:"job_name"`
+		StaticConfigs []yamlStaticConfig `yaml:"static_configs"`
+		MetricsPath   string             `yaml:"metrics_path"`
+	}
+
+	type yamlConfig struct {
+		Global struct {
+			ScrapeInterval string `yaml:"scrape_interval"`
+			ScrapeTimeout  string `yaml:"scrape_timeout"`
+		}
+		ScrapeConfigs []yamlScrapeConfig `yaml:"scrape_configs"`
+	}
+
+	cfg := yamlConfig{}
+	cfg.Global.ScrapeInterval = "10s"
+	cfg.Global.ScrapeTimeout = "5s"
+
+	for _, scrapeConfig := range scrapeConfigs {
+		var targets []string
+		for _, scrapeNode := range scrapeConfig.ScrapeNodes {
+			ips, err := c.ExternalIP(ctx, scrapeNode.Nodes)
+			if err != nil {
+				return "", err
+			}
+			for _, ip := range ips {
+				targets = append(targets, fmt.Sprintf("%s:%d", ip, scrapeNode.Port))
+			}
+		}
+
+		cfg.ScrapeConfigs = append(
+			cfg.ScrapeConfigs,
+			yamlScrapeConfig{
+				JobName:     scrapeConfig.JobName,
+				MetricsPath: scrapeConfig.MetricsPath,
+				StaticConfigs: []yamlStaticConfig{
+					{
+						Targets: targets,
+					},
+				},
+			},
+		)
+	}
+
+	ret, err := yaml.Marshal(&cfg)
+	return string(ret), err
+}
+
+// MakeWorkloadScrapeConfig creates a scrape config for a workload.
+func MakeWorkloadScrapeConfig(jobName string, scrapeNodes []ScrapeNode) ScrapeConfig {
+	return ScrapeConfig{
+		JobName:     jobName,
+		MetricsPath: "/",
+		ScrapeNodes: scrapeNodes,
+	}
+}
+
+// MakeInsecureCockroachScrapeConfig creates scrape configs for the given
+// cockroach nodes. All nodes are assumed to be insecure and running on
+// port 26258.
```

Comment:
I believe there is a way to get the port via roachprod if this function were to take a *cluser as well.  But, I think that will com naturally if we decide to follow-up with local-mode support.